src/query/functions/src/scalars/decimal.rs

```diff
@@ -468,13 +473,72 @@ fn convert_to_decimal(
     dest_type: DataType,
 ) -> Value<AnyType> {
     let arg = &args[0];
-    if from_type.is_integer() {
-        return integer_to_decimal(arg, ctx, from_type, dest_type);
+    match from_type {
+        DataType::Number(ty) => {
+            if ty.is_float() {
+                float_to_decimal(arg, ctx, from_type, dest_type)
+            } else {
+                integer_to_decimal(arg, ctx, from_type, dest_type)
+            }
+        }
+        DataType::Decimal(_) => decimal_to_decimal(arg, ctx, from_type, dest_type),
+        DataType::String => string_to_decimal(arg, ctx, from_type, dest_type),
+        _ => unreachable!("to_decimal not support this DataType"),
     }
-    if from_type.is_floating() {
-        return float_to_decimal(arg, ctx, from_type, dest_type);
+}
+
+fn string_to_decimal_internal<T: Decimal>(
+    ctx: &mut EvalContext,
+    string_column: &StringColumn,
+    size: DecimalSize,
+    dest_type: &DecimalDataType,
+) -> DecimalColumn {
+    let mut builder = DecimalDeserializer::<T>::with_capacity(dest_type, string_column.len());
```

Comment:
better to use `DecimalColumnBuilder`.

Reply:
ok

Reply:
This way we can use `Vec<T>` to store the decimal value directly and the `DecimalDeserializer` is not needed

Reply:
This way we can use `Vec<T>` to store the decimal value directly and the `DecimalDeserializer` is not needed

Reply:
This way we can use `Vec<T>` to store the decimal value directly and the `DecimalDeserializer` is not needed